### PR TITLE
konnect: use sdk-konnect-go for ControlPlanes

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

Remove manually maintained ControlPlane client and use [sdk-konnect-go](https://github.com/Kong/sdk-konnect-go/) for that.

**Which issue this PR fixes**:

Part of #6548.

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: hack/cleanup/konnect_runtime_groups.go [hunk 1]
```diff
@@ -4,42 +4,37 @@ import (
...
+	"io"
 	"net/http"
 	"time"
 
 	"github.com/go-logr/logr"
-	"github.com/oapi-codegen/runtime/types"
 	"github.com/samber/lo"
 
-	cp "github.com/kong/kubernetes-ingress-controller/v3/internal/konnect/controlplanes"
+	sdkkonnectgo "github.com/Kong/sdk-konnect-go"
```
Comment (randmonkey) [inline hack/cleanup/konnect_runtime_groups.go:14]: Should the packages from `sdk-konnect-go` have a separate section in imports?

File: internal/manager/metadata/metadata.go [hunk 2]
```diff
@@ -3,8 +3,16 @@ package metadata
...
 import (
 	"strings"
+
+	"github.com/Kong/sdk-konnect-go/pkg/metadata"
 )
 
+func init() {
+	// NOTE: We do it this way because speakeasy does not provide a way to set the
+	// user-agent for the SDK instance.
+	metadata.SetUserAgent(UserAgent())
```
Comment (randmonkey) [inline internal/manager/metadata/metadata.go:13]: The imported package name `metadata` looks confusing in another package named `metadata`. How about renaming the `github.com/Kong/sdk-konnect-go/pkg/metadata` to `sdkkonnectmetadata`?

**Review by randmonkey — COMMENTED** [review]
Generally LGTM, some minor comments.
Also a upper/lower case problem: The currently used packages from `kong` are using lower case `kong`. Any reason for the PR to use `Kong`?

File: hack/cleanup/konnect_runtime_groups.go (hunk 1 above)
Comment (pmalek) [reply hack/cleanup/konnect_runtime_groups.go:14]: IMHO yes, that's why I made it this way here and added a rule in the linter: https://github.com/Kong/kubernetes-ingress-controller/blob/faedca400fe43f6dce48540e5359f4fbee475a7b/.golangci.yaml#L57.

WDYT?

File: internal/manager/metadata/metadata.go (hunk 2 above)
Comment (pmalek) [reply internal/manager/metadata/metadata.go]: 👍 

added in cf3274c710af547ed0d9025572f9ccf7c90b6186

File: hack/cleanup/konnect_runtime_groups.go
Comment (czeslavo) [inline hack/cleanup/konnect_runtime_groups.go:1]: Let's rename this file to `konnect_control_planes.go` for consistency.

File: internal/konnect/sdk/sdk.go [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package sdk
+
+import (
+	sdkkonnectgo "github.com/Kong/sdk-konnect-go"
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+)
+
+// SDK returns a new SDK instance.
+func SDK(token string, opts ...sdkkonnectgo.SDKOption) *sdkkonnectgo.SDK {
```
Comment (czeslavo) [inline internal/konnect/sdk/sdk.go:9]: Maybe `sdk.New` will make more sense than `sdk.SDK`, WDYT?
```suggestion
func New(token string, opts ...sdkkonnectgo.SDKOption) *sdkkonnectgo.SDK {
```

**Review by czeslavo — COMMENTED** [review]
LGTM 👍

File: internal/konnect/sdk/sdk.go (hunk 3 above)
Comment (pmalek) [reply internal/konnect/sdk/sdk.go]: bc5d7a6773e5f10c86c2e2716716a599c65b0a55

File: go.mod [hunk 4]
```diff
@@ -76,12 +77,12 @@ require (
 	github.com/Azure/go-ansiterm v0.0.0-20210617225240-d185dfc1b5a1 // indirect
 	github.com/Kong/go-diff v1.2.2 // indirect
 	github.com/Kong/gojsondiff v1.3.2 // indirect
+	github.com/Kong/sdk-konnect-go v0.1.5
```
[the remark this answers is not in the document]
Comment (pmalek) [reply go.mod:80]: After having the poll at https://kongstrong.slack.com/archives/C011RQPHDC7/p1730975025432279 let's leave this as is.

**Review by czeslavo — APPROVED** [review]

**Pull request merged**